src/main/java/ulcrs/controllers/SessionController.java

```diff
@@ -14,36 +25,82 @@
 
 public class SessionController extends BaseController {
 
+    private static final String WORKSPACE_PATH = "workspace/";
+
     @Override
     public RouteGroup routes() {
         return () -> {
             before("/*", (request, response) -> log.info("endpoint: " + request.pathInfo()));
-            get("/", this::getSessionList, gson::toJson);
+            get("/", this::getSessionList, sessions -> {
+                Gson gson = new GsonBuilder().excludeFieldsWithoutExposeAnnotation().setPrettyPrinting().create();
+                return gson.toJson(sessions);
+            });
             get("/:name", this::getSession, gson::toJson);
-            post("/:name", this::updateSession, gson::toJson);
+            post("/:name", this::saveSession, gson::toJson);
             delete("/:name", this::deleteSession, gson::toJson);
         };
     }
 
     private List<String> getSessionList(Request request, Response response) {
-        // TODO: return names of sessions with limit specified
-        return null;
+        response.type(CONTENT_TYPE_JSON);
+        int limit = Integer.valueOf(request.params("limit"));
+
+        List<String> filenames = new ArrayList<>();
+        File folder = new File(WORKSPACE_PATH);
+        File[] listOfFiles = folder.listFiles();
+        if (listOfFiles != null) {
+            for (int i = 0; i < limit && i < listOfFiles.length; i++) {
```

Comment:
How are these files sorted? Does listFiles() return them in an array according to name, most recently accessed, most recently saved, or something else? I would think we would want to return the most recently saved (i.e. most recently used) sessions.

Reply:
Currently they are designed to be stored using epoch seconds. I'll change them to be listed backwards